src/main/java/com/istresearch/galton/MetricsAppender.java

```diff
@@ -0,0 +1,362 @@
+     *  This will provide an option that doesn't require client's to use the LogstashMarker.
+     */
+    @Override
+    protected void append(final ILoggingEvent event) {
+        Marker marker = event.getMarker();
+        Map<String, Object> markerMap = convertMarkerToMap(marker);
+        String logMsg = event.getMessage().toLowerCase().replace(" ", "_");
+
+        //checks metric config to see if logMsg was used for a counter
+        if(getCounters().containsKey(logMsg)) {
+            Metric m = getCounters().get(logMsg);
+            if(markerMap.containsKey("tags")) m.addTags((Map<String, String>)markerMap.get("tags"));
+            publishCounterMetric(m);
+        }
+        //checks metric config to see if logMsg was used for a gauge
+        if(getGauges().containsKey(logMsg)) {
+            Metric m = getGauges().get(logMsg);
+            if(markerMap.containsKey("tags")) m.addTags((Map<String, String>)markerMap.get("tags"));
+            publishGaugeMetric(m, Integer.valueOf(String.valueOf(markerMap.get(m.getMarkerKey()))));
+        }
+        for(String key : markerMap.keySet()) {
+            //checks each key to see if it contains the prefix
+            if(key.contains(getCounterPrefix())) {
+                Metric m = new Metric();
+                m.setMetricName(String.valueOf(markerMap.get(key)));
+                if(markerMap.containsKey("tags")) m.addTags((Map<String, String>)markerMap.get("tags"));
+                publishCounterMetric(m);
+            }
+            else if(key.contains(getGaugePrefix())) {
+                Metric m = new Metric();
+                m.setMarkerKey(key.replace(getGaugePrefix(), ""));
+                if(markerMap.containsKey("tags")) m.addTags((Map<String, String>)markerMap.get("tags"));
+                publishGaugeMetric(m, Integer.valueOf(String.valueOf(markerMap.get(key))));
+            }
+            //checks to see if the marker key was used in the metric config (requires logMsg to be absent for this metric).
+            if(getCounters().containsKey(key)) {
+                if(markerMap.containsKey("tags")) getCounters().get(key).addTags((Map<String, String>)markerMap.get("tags"));
+                publishCounterMetric(getCounters().get(key));
+            }
+            if(getGauges().containsKey(key)) {
+                if(markerMap.containsKey("tags")) getGauges().get(key).addTags((Map<String, String>)markerMap.get("tags"));
+                publishGaugeMetric(getGauges().get(key), Integer.valueOf(String.valueOf(markerMap.get(key))));
+            }
+        }
+    }
+
+    /**
+     * Converts the Logstash Marker into a HashMap<String, Object>
+     * @param marker
+     * @return Map
+     */
+    public static Map<String, Object> convertMarkerToMap(Marker marker) {
+        Map<String, Object> markerMap = new HashMap<String, Object>();
+        if(marker == null) return markerMap;
```

Comment:
Recommend returning `Collections.emptyMap()` and initializing `markerMap = null;` If the marker isn't null, then the map just gets tossed because the map from the MarkerMapGenerator is used, if the marker is null the map returned is presumably not ever used. It's a nit.

Reply:
Good recommendation, I refactored.